### PR TITLE
🔥 🔥 Working version

### PR DESCRIPTION
**Pull request by Adhikram**

### CODE REVIEW

File: devsnest/app/workers/group_modifier_worker.rb [hunk 1]
```diff
@@ -4,8 +4,18 @@
 class GroupModifierWorker
   include Sidekiq::Worker
   sidekiq_options retry: 2
-  def perform(action, group_name)
+  def perform(action, group_name, server_guild_id = nil)
+    group = Group.find_by(name: group_name[0])
+    guild_id = if server_guild_id.present?
```
Comment (Yug-gurnani) [inline devsnest/app/workers/group_modifier_worker.rb:9]: similar blocks of code found at many locations, make a function and call it on all places..

File: devsnest/config/routes.rb [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # frozen_string_literal: true
 
 Rails.application.routes.draw do
+  resources :servers
```
Comment (Yug-gurnani) [inline devsnest/config/routes.rb:4]: use json_api_resources

**Review by Yug-gurnani — DISMISSED** [review]

File: devsnest/app/controllers/api/v1/users_controller.rb [hunk 3]
```diff
@@ -319,6 +319,14 @@ def upload_i_have_enrolled_for_course_image
         render_success({ message: 'Image uploaded successfully' })
       end
 
+      def check_group_name
+        discord_id = params['data']['attributes']['discord_id']
+        user = User.find_by(discord_id: discord_id)
+        groupmember = GroupMember.find_by(user_id: user.id)
```
[the remark this answers is not in the document]
Comment (Adhikram) [reply devsnest/app/controllers/api/v1/users_controller.rb:325]: Done

File: devsnest/app/models/group.rb [hunk 4]
```diff
@@ -23,6 +25,10 @@ class Group < ApplicationRecord
   def parameterize
     update_attribute(:slug, name.parameterize)
   end
+  def assign_server
+    server = Server.find_by(id: ENV['SERVER_ID'])
+    update_attribute(:server_id, server.id) if server.present?
```
[the remark this answers is not in the document]
Comment (Adhikram) [reply devsnest/app/models/group.rb]: Done

File: devsnest/app/workers/group_modifier_worker.rb (hunk 1 above)
Comment (Adhikram) [reply devsnest/app/workers/group_modifier_worker.rb]: Done

File: devsnest/config/routes.rb (hunk 2 above)
Comment (Adhikram) [reply devsnest/config/routes.rb]: Done

**Review by Yug-gurnani — CHANGES_REQUESTED** [review]

**Review by Yug-gurnani — APPROVED** [review]

**Pull request merged**